File: src/Models/LetsEncryptServer/Account/Responses/GetAccountResponse.cs
using MaksIT.Core.Abstractions.Webapi;


namespace MaksIT.Models.LetsEncryptServer.Account.Responses;

public class GetAccountResponse : ResponseModelBase {
  public Guid AccountId { get; set; }
  public required bool IsDisabled { get; set; }
  public string? Description { get; set; }
  public required string[] Contacts { get; set; }
  public string? ChallengeType { get; set; }
  public GetHostnameResponse[]? Hostnames { get; set; }
  public required bool IsStaging { get; set; }
}

File: src/ClientApp/models/letsEncryptServer/account/responses/GetAccountResponse.ts
import { CacheAccount } from '@/entities/CacheAccount'
import { GetHostnameResponse } from './GetHostnameResponse'

export interface GetAccountResponse {
  accountId: string
  isDisabled: boolean
  description: string
  contacts: string[]
  challengeType?: string
  hostnames?: GetHostnameResponse[]
  isStaging: boolean
}

const toCacheAccount = (account: GetAccountResponse): CacheAccount => {
  return {
    accountId: account.accountId,
    isDisabled: account.isDisabled,
    description: account.description,
    contacts: account.contacts.map((contact) => contact),
    challengeType: account.challengeType,
    hostnames:
      account.hostnames?.map((hostname) => ({
        hostname: hostname.hostname,
        expires: new Date(hostname.expires),
        isUpcomingExpire: hostname.isUpcomingExpire,
        isDisabled: hostname.isDisabled
      })) ?? [],
    isStaging: account.isStaging,
    isEditMode: false
  }
}

export { toCacheAccount }
